docs/ql-documentation/learn-ql/cpp/introduce-libraries-cpp.rst

```diff
@@ -4,114 +4,516 @@ Introducing the C/C++ libraries
+There is an extensive QL library for analyzing snapshots extracted from C/C++ projects. The QL classes in this library present the data from a snapshot database in an object-oriented form and provide abstractions and predicates to help you with common analysis tasks.  The library is implemented as a set of QL modules, that is, files with the extension ``.qll``. The module ``cpp.qll`` imports all of the core C/C++ library modules, so you can include the complete library by beginning your query with:
 
 .. code-block:: ql
 
    import cpp
 
-The rest of this topic briefly summarizes the most important QL classes and predicates provided by this library.
+The rest of this topic summarizes available QL classes and corresponding C/C++ constructs.
 
-   You can find related classes and features using the query console's auto-complete feature. You can also press **F3** to jump to the definition of any element (QL library files are opened in new tabs in the console).
+NOTE: You can find related classes and features using the query console's auto-complete feature.  You can also press **F3** to jump to the definition of any element; QL library files are opened in new tabs in the console.
 
 Summary of the library classes
-------------------------------
+==============================
+
+The most commonly used standard QL library classes are listed below.  The listing is broken down by functionality.  Each QL library class is annotated with a C/C++ construct it corresponds to.
+
+Declaration classes
+-------------------
+
+This table lists `Declaration <https://help.semmle.com/qldoc/cpp/semmle/code/cpp/Declaration.qll/type.Declaration$Declaration.html>`__ classes representing C/C++ declarations.
+
++---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Comment:
Would the table render the same way if each of these long lines is changed to `+--+--+--+`? It would make it a lot easier to view the file in GitHub's diff viewer.

Reply:
Sadly not. The `---+` separators need to be the the full width of the text.

@zlaski-semmle: I'd recommend rebasing to fix the conflicts so that you can preview the sphinx-generated output. The rendered table would be much easier to read.

Reply:
I  believe that the conflict has now been resolved (the file was moved to a new location).